shotover-proxy/src/tcp.rs

```diff
@@ -0,0 +1,19 @@
+use anyhow::{anyhow, Result};
+use std::time::Duration;
+use tokio::{
+    net::{TcpStream, ToSocketAddrs},
+    time::timeout,
+};
+
+pub async fn tcp_stream<A: ToSocketAddrs + std::fmt::Debug>(destination: A) -> Result<TcpStream> {
+    timeout(Duration::from_secs(3), TcpStream::connect(&destination))
```

Comment:
Given this is a refactor, we should probably address making this timeout configurable with a 3 sec default. I can't imagine why we would need to longer, but shorter might be appropriate for some cases.

Reply:
That is quite a large change so I've done it in a separate PR with this PR as a prereq: https://github.com/shotover/shotover-proxy/pull/901